libraries/chain/include/steem/chain/smt_objects/smt_token_object.hpp

```diff
@@ -172,6 +187,42 @@ class smt_event_token_object : public object< smt_event_token_object_type, smt_e
    time_point_sec                launch_expiration_time;
 };
 
+class smt_contribution_object : public object< smt_contribution_object_type, smt_contribution_object >
+{
+   STEEM_STD_ALLOCATOR_CONSTRUCTOR( smt_contribution_object );
+
+public:
+   template< typename Constructor, typename Allocator >
+   smt_contribution_object( Constructor&& c, allocator< Allocator > a )
+   {
+      c( *this );
+   }
+
+   id_type                               id;
+   asset_symbol_type                     symbol;
+   account_name_type                     contributor;
+   uint32_t                              contribution_id;
+   asset                                 contribution;
+};
+
+struct by_symbol_contributor;
+
+typedef multi_index_container <
+   smt_contribution_object,
+   indexed_by <
+      ordered_unique< tag< by_id >,
+         member< smt_contribution_object, smt_contribution_object_id_type, &smt_contribution_object::id > >,
+      ordered_unique< tag< by_symbol_contributor >,
```

Comment:
Add a non-consensus order for a full node that sorts by `{contributor, symbol, constribution_id}` so that UIs can quickly show users all the ICOs they have contributed to. Also add this to order to the API.